src/View/ViewBuilder.php

```diff
@@ -301,25 +301,64 @@ public function getPlugin(): ?string
     }
 
     /**
-     * Adds a helper to use.
+     * Adds a helper to use if no such helper with this name already exists.
      *
      * @param string $helper Helper to use.
      * @param array<string, mixed> $options Options.
      * @return $this
+     * @throws \RuntimeException When a duplicate is found.
      * @since 4.1.0
      */
     public function addHelper(string $helper, array $options = [])
     {
-        $this->_helpers[$helper] = $options;
+        [$plugin, $name] = pluginSplit($helper);
+        if (isset($this->_helpers[$name])) {
+            $message = sprintf('The `%s` alias has already been added as helper. Use `setHelper()` instead.', $name);
+
+            throw new RuntimeException($message);
```

Comment:
```suggestion
            throw new CakeException($message);
```

Reply:
This was copied from registry 
Should the other parts of the System also use a similar exception moving forward?